wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -187,7 +227,179 @@ export default {
           }
         }
       })(_this);
-    }
+    },
+
+    /**
+     * This function gets executed after fetching posts data to set the Posts array.
+     * We are initializing the pagination array.
+     **/ 
+
+    process: function(data, query, headers) {
+      // If there are posts for this query, map them to the template.
+      let posts = (Array.isArray(data)) ?
+        data.map(this.maps()[this.type]) : false;
+
+      /**
+       * Set only current page posts
+       * The query argument gets the current, previous, next page
+       * So, we set Posts array with current page posts
+       */
+      
+      if(this.query.page == query.page){
```

Comment:
Yes, this executes only once after component is rendered initially and set posts of the page 1. I will update it with if(this.query.page == 1), instead of validating with query.page.

Reply:
Can you make this update?